ezmlm-issubn.1
.\" $Id: ezmlm-issubn.1 170 2005-01-04 16:08:10Z bruce $
.TH ezmlm-issubn 1
.SH NAME
ezmlm-issubn \- test to see if an address is subscribed to a mailing list
.SH SYNOPSIS
.B ezmlm-issubn
[
.B \-nN
]
.I dir
.I [ dir1 ... ]
.SH DESCRIPTION
.B ezmlm-issubn
checks to see if the address obtained from the environment variable
.I SENDER
is subscribed to the mailing list stored in
.I dir 
or the mailing list in
.I dir1 
or ...

If
.I SENDER
is not defined
.B ezmlm-issubn
exits with an error.

If
.I SENDER
is on [any of] the mailing list[s],
.B ezmlm-issubn
exits with a zero exit code.

If
.I box\fB@\fIdomain
is not on the mailing list,
.B ezmlm-issubn
exits 99. This exit code is non-success from a shell point of view, but to
qmail it means "success and skip remaining lines in the .qmail file). Thus,
a simple way to execute a delivery if the
.ezmlm-issubn
criteria are met is to place the
.B ezmlm-issubn
line first, followed by the action line. If SENDER is a subscriber, the
action line is executed, if not, the line is ignored without the generation
of an error condition. To generate a fatal error, just:

.EX
|/path/ezmlm-issubn
.I dir1 ...
|| (echo "err msg"; exit 100)

|/path/action_for_subscribers
|/path/more_for_subscribers
.EE

.B ezmlm-issubn
exits 100 on permanent and 111 on temporary errors.
.SH OPTIONS
.TP
.B \-n
Negate exit code. Exit 99 if SENDER is a subscriber and exit 0 if not.
This is useful when trying to exclude SENDERs.
.TP
.B \-N
(Default.)
Normal exit codes: 0 is the address is in any of the lists, 99 if not.
.SH "SEE ALSO"
ezmlm-list(1),
ezmlm-manage(1),
ezmlm-make(1),
ezmlm-send(1),
ezmlm-sub(1),
ezmlm-unsub(1),
ezmlm(5)


CHANGES
BUGS:	-ezmlm-idx does not implement a full rfc822 parser. Comments can be
	 inserted to defeat the MIME parsing mechanism, such as:
	 Content-Type: (comment) test/plain
	 This type of comment, while legal, does not occur in normal messages.
	 Guiding info past ezmlm-idx MIME parsing is easier done by enclosing
	 a message within a message. The mechanism isn't intended as a
	 "secure" way of preventing e.g. "application/ms-tnef" from being
	 disseminated.
	 If you think this is a problem, E-mail ezmlm@list.cr.yp.to.
	 Mimeremove/reject work only at the outer MIME layer.
	-ezmlm-request requires addresses to be user@host. While rfc822
	 compliant, addresses like ``"the man" @ host'' are not accepted.
	-ezmlm-cgi doesn't deal optimally with holes in the archive.
	-ezmlm-cgi seems not to work with Chinese. The problem has been to
	 find someone willing/able to participate in debugging.

Some ezmlmrc are not updated and may not work right.  If you fix them,
please send them to ezmlm@list.cr.yp.to for inclusion in the next
release.

- Added <#C#>, <#T#>, and <#X#> substitutions for (un)subscribe
  confirmation emails, replaced with the hash code, time stamp, and
  action respectively.

- Added option to ezmlm-[un]sub to use a tag other than "manual" in Log.

- Added option to ezmlm-manage to show what response is being sent.

- Fixed behavior of ezmlm-manage -Q flag to match man page.

- Fixed ezmlm-import failing to flush output to the last message,
  and enhanced it to allow reading the mbox from stdin.
  Thanks to Tullio Andreatta.

- Changed to UTF-8

ezmlm-idx-7.1.1, 2010-11-18
===========================

- Fixed bug in getln2 function that prevented error handling.

- Fixed bug in ezmlm-cron that prevented parsing of spaces.

- Fixed unclosed file descriptor on error path in ezmlm-cgi.

ezmlm-idx-7.1.0, 2010-11-03
===========================

- Added support for SQLite3 subscriber databases.
  Thanks to Mike Tedder.

- Added support for wildcard addresses in all address databases.

- Added support for qmqpservers to all ezmlm-idx programs.

- Added support for decoding sender addresses mangled with a BATV "btv1" tag.

- Added support for custom subject lines to ezmlm-get (for digests).

- Added a new "omitbottom" control file, equivalent to the -B option for
  both ezmlm-get and ezmlm-manage.

- Modified the (un)subscribe procedure to not require confirmation
  before telling the sender they were already on (or off) the list.

- Renamed to "setup" makefile target to the more standard "install".

- Modified ezmlm-split to exit silently if there is not "split" file,
  and added it back into the "manager" file to properly handle subscribe
  and unsubscribe requests destined for sublists.

- Made the ezmlm-reject -h (obey headerreject) option the default.

- Fixed ezmlm-send to obey the "addtrailer" setting instead of looking
  for a "text/trailer" file.

- Fixed handling of adding the trailer on old lists.

- Fixed adding the trailer to posts encoded with base64.

- Fixed handling Received: headers with the date stamp on a separate line.

- Fixed bug in command-line option processing which caused sender-
  confirmed posting to fail.

- Fixed handling BATV sender addresses with upper-case hex tags.
  Thanks to Kyle Wheeler for pointing this out.

- Adjusted filename in digest attachments to fix problem with Outlook.
  Thanks to Glen Stewart

ezmlm-idx-7.0.2, 2009-08-09
===========================

- Fixed selection of [un]subscribe subject line for moderated
  subscriptions.

- Clarified ezmlm-reject man page to note that Precedence: bulk is also
  rejected.

- Added a proper charset for ch_GB (Chinese GuoBiao), and added an
  explicit charset for the other languages that were missing them.

- Switched all ISO-8859-1 charsets to the more modern ISO-8859-15.

- Many small text updates and tweaks.

ezmlm-idx-7.0.1, 2009-07-09
===========================

- Added support for decoding sender addresses mangled by BATV prvs.

- Added missing ezmlm-checksub to the installed programs.

- Improved the TXT_BY message used in digests to work better with
  non-English languages.

- Fixed out-of-memory resulting in a permanent error.

- Fixed a buffer underflow bug in concatHDR.

- Fixed a bug in ezmlm-issubn.c that was causing it to scan the main
  list when it shouldn't.

- Fixed missing defaults in ezmlm-tstdig when digsize, digcount, or
  digtime were not created.

- Clarified the steps necessary when upgrading in UPGRADE.

ezmlm-idx-7.0.0, 2008-06-16
===========================

This version has three major architectural changes:

1. There is now support for internationalized messages.  All error
messages, all subject lines, and a few other messages are now configured
in a file named "text/messages" that is read at run time.  Messages in
this file will undergo header-style substitution before they are output.

Unlike other text files, all 3 of the files (DIR/text/messages,
/etc/ezmlm/LANG/text/messages, and /etc/ezmlm/default/text/messages) are
read, and only the first match (in the above order) is used for any
given message.  This allows for creation of partial files to override
just select messages.  If no match is found, the internal English text
is used.

2. The use of the "flags" file has been deprecated completely by
individual flag files accessed by the appropriate programs.  This, along
with corresponding changes to the ezmlmrc template, allows for lists to
be reconfigured without invoking ezmlm-make.  To upgrade a list to the
new setup, simply run "ezmlm-make -+ DIR".  Note that this will delete
the "config" and "flags" files, as they have been superceded by other
files.

3. Creation of subscriber tables has been moved out of ezmlm-mktab-* and
into ezmlm-make by adding some additional hooks to the subdb plugins.  A
ezmlm-rmtab program is also added to remove subscriber tables, also with
hooks in the subdb plugins.  The ezmlm-mktab-* programs have been

ezmlm-issubn.c
/*$Id: ezmlm-issubn.c 421 2005-09-14 05:46:15Z bruce $*/

#include <unistd.h>
#include "strerr.h"
#include "env.h"
#include "subscribe.h"
#include "sgetopt.h"
#include "errtxt.h"
#include "die.h"
#include "idx.h"
#include "config.h"
#include "auto_version.h"

const char FATAL[] = "ezmlm-issubn: fatal: ";
const char USAGE[] =
"ezmlm-issubn: usage: ezmlm-issubn [-nN] dir [dir1 ...]";

void main(int argc,char **argv)
{
  char *dir;
  char *addr;
  int flagsub = 0;
  int opt;

  addr = env_get("SENDER");
  if (!addr) die_sender();	/* REQUIRE sender */

  while ((opt = getopt(argc,argv,"nNvV")) != opteof)
    switch(opt) {
      case 'n': flagsub = 99; break;
      case 'N': flagsub = 0; break;
      case 'v':
      case 'V': strerr_die2x(0, "ezmlm-issubn version: ",auto_version);
      default:
	die_usage();
    }

  startup(argv[optind]);

  while ((dir = argv[optind++])) {
    if (dir[0] != '/')
      strerr_die2x(100,FATAL,ERR_SLASH);
    if (issub(dir,0,addr)) {
      closesub();
      _exit(flagsub);		/* subscriber */
    }
  }
  closesub();
  if (flagsub)			/* not subscriber anywhere */
    _exit(0);
  else
    _exit(99);
}
